### DISCUSSION TITLE
@socket.io/redis-stream-adapter not forwarding message to client?

### DISCUSSION THREAD

**Discussion by shubham-thorat** [socketio/socket.io#4837] [Q&A]
Hi everyone,
I am trying to create simple producer and consumer and client using socket.io && socket.io/redis-stream-adapter
On producer side : I am adding to redis using node-redis xadd method to 'mystream'
On consumer side : creating server as mention here  https://socket.io/docs/v4/redis-streams-adapter/ on specified stream as 'mystream'
On client: connected to server(consumer) using socket.io

I am able see to the see message added using xadd on redis and consumer able consume it but not able send 
client. Note: client is connected to server(consumer) and other normal emit message able to consume but not 
forwarding producers message using redis-streams-adapter to client.
What can be the issue? (using nodejs)

**Comment by darrachequesne** [comment]
Here is the expected format for the data in the stream: https://github.com/socketio/socket.io-redis-streams-adapter/blob/93e28198ec85e595444b8fcb522d930bca67c5aa/lib/adapter.ts#L158-L182

I think we will have to implement a proper emitter, like the one existing for Redis: https://socket.io/docs/v4/redis-adapter/#emitter
